engines/registry/lib/registry/engine.rb

```diff
@@ -2,5 +2,17 @@ module Registry
   class Engine < ::Rails::Engine
     isolate_namespace Registry
     config.generators.api_only = true
+
+    config.after_initialize do
+      Api::Connect::V3::Systems::ActivationsController.class_eval do
+        before_action :handle_auth_cache, only: %w[index]
+
+        def handle_auth_cache
+          unless ZypperAuth.verify_instance(request, logger, @system, registry: true)
```

Comment:
It _feels_ like we've overloaded `ZypperAuth` here, having it verify authN for a registry.

IMO it's out of scope for this PR, but it seems that `#verify_instance` should be in `InstanceVerification`, alongside `#update_cache`. As a bonus; the key pattern could be written _once_ in one function, instead of being inline in `InstanceVerification#update_cache` and `ZypperAuth#verify_instance`.

Reply:
Why would we care in the authentication path that the full validation happens because of registry access? I do not understand. That just appears to be completely incorrect. A full system validation means we validate the metadata we receive, that is instance data must match the product data. If everything matches we put the system in the cache. So why is the path how we verify the instance important?

Reply:
GitHub still showed me the `registry` argument, if it's gone then we are good to go.